creator-node/src/services/stateMachineManager/stateReconciliation/SecondarySyncHealthTracker.ts

```diff
@@ -1,332 +1,284 @@
- */
-async function _getMetricsMatchingPattern(
-  pattern: string,
-  wallets: string[] = []
-) {
-  const keys: string[] = wallets
-    ? await _getAllKeysMatchingPattern(pattern, (key) =>
-        wallets.some((wallet) => key.includes(wallet))
-      )
-    : await _getAllKeysMatchingPattern(pattern)
+  /**
+   * Given user info, generate the redis keys
+   * @param userInfo user info
+   * @returns the redis keys
+   */
+  async _getSyncFailureKeyForWalletOnSecondaries(
+    userInfo: ComputeWalletOnSecondaryUserInfo[]
+  ): Promise<string[]> {
+    const syncFailureKeys = []
+    for (const { wallet, secondary1, secondary2 } of userInfo) {
+      const redisKeyWalletToSecondary1 = this._getSyncFailureRedisKey({
+        secondary: secondary1,
+        wallet,
+        syncType: '*'
+      })
 
-  // Short-circuit here since redis `mget` throws if array param has 0-length
-  if (!keys || !keys.length) {
-    return {}
-  }
+      const redisKeyWalletToSecondary2 = this._getSyncFailureRedisKey({
+        secondary: secondary2,
+        wallet,
+        syncType: '*'
+      })
 
-  // This works because vals.length === keys.length
-  // https://redis.io/commands/mget
-  const vals = await redisClient.mget(keys)
+      syncFailureKeys.push(redisKeyWalletToSecondary1)
+      syncFailureKeys.push(redisKeyWalletToSecondary2)
+    }
 
-  // Zip keys and vals arrays into map of key-val pairs
-  const keyMap: { [key: string]: any } = {}
-  for (let i = 0; i < keys.length; i++) {
-    keyMap[keys[i]] = vals[i]
+    return syncFailureKeys
   }
 
-  return keyMap
-}
+  _determineIfSyncShouldReenqueue({
+    wallet,
+    secondary,
+    errorToErrorCount
+  }: {
+    wallet: string
+    secondary: string
+    errorToErrorCount: { [error: string]: number /* the error count */ }
+  }) {
+    // Get the error and number of times the error occurred
+    for (const [error, errorCount] of Object.entries(errorToErrorCount)) {
+      // Compare the above number to the max retry attempts
+      const shouldContinueAction = this._shouldContinueAction(error, errorCount)
+
+      if (!shouldContinueAction) {
+        this._updateWalletToSecondaryAndMaxErrorReached({
+          wallet,
+          secondary,
+          error
+        })
+      }
 
-/**
- * Returns array of all keys in Redis matching pattern, using redis SCAN
- * https://github.com/luin/ioredis#streamify-scanning
- *
- * @returns array | Error
- */
-async function _getAllKeysMatchingPattern(
-  pattern: string,
-  extraFilter = (_: string) => true
-): Promise<string[]> {
-  const stream = redisClient.scanStream({ match: pattern })
-
-  const keySet = new Set<string>()
-  return new Promise<string[]>((resolve, reject) => {
-    stream.on('data', async (keys = []) => {
-      keys.filter(extraFilter).forEach((key) => {
-        keySet.add(key)
+      // If under, record retry to 'true'. Else, record retry as 'false'.
+      this._updateWalletToSecondaryToShouldContinueAction({
```

Comment:
can this take in a mapping and return the updated mapping instead of updating the class object (`this.walletToSecondaryToShouldContinueAction`)? that seems like the main blocker to making this not be a class

Reply:
yup, basically rewrote this to something of that essence!